src/translator/html.cpp

```diff
@@ -108,10 +120,37 @@ bool intersects(ByteRange const &range, HTML::Span const &span) {
   return range.begin <= span.end && range.end >= span.begin;
 };
 
-bool containsTag(HTML::Taint const &stack, HTML::Tag const *tag) {
+bool contains(HTML::TagNameSet const &set, std::string_view const &name) { return set.find(name) != set.end(); }
+
+bool contains(HTML::TagStack const &stack, HTML::Tag const *tag) {
   return std::find(stack.rbegin(), stack.rend(), tag) != stack.rend();
 }
 
+bool isSubset(HTML::TagStack const &a, HTML::TagStack const &b) {
+  if (a.size() > b.size()) return false;
+
+  for (auto i = a.begin(), j = b.begin(); i != a.end(); ++i, ++j)
+    if (*i != *j) return false;
+
+  return true;
+}
+
+template <typename T>
+size_t argmax(std::vector<T> const &items) {
+  assert(!items.empty());
+  size_t best = 0;
+  for (size_t i = 1; i + 1 < items.size(); ++i) {
+    if (items[i] > items[best]) {
+      best = i;
+    }
+  }
+  return best;
+}
```

Comment:
I searched for argmax, didn't find it, assumed it wasn't in the standard library. Using max_element now (although the call site hasn't gotten any prettier because of it :( )